source/tyr/tyr/resources.py

```diff
@@ -701,3 +726,69 @@ def delete(self, name=None, traveler_type=None):
         except Exception:
             logging.exception("fail")
             raise
+
+
+class Plan(flask_restful.Resource):
+    def __init__(self):
+        pass
+
+    def get(self, plan_id=None):
+        if plan_id:
+            plan = models.Plan.query.get_or_404(plan_id)
+            return marshal(plan, plan_fields)
+        else:
+            plans = models.Plan.query.all()
+            return marshal(plans, plan_fields)
```

Comment:
I don't think it's working, you have a list that you try to marshall as a single object